Setup docs

SarahAlidoost opened the pull request:
closes #32  
closes #33 

To view the documentation page locally:
```
pip install .[docs]
mkdocs serve
```
Click on the localhost link.

fnattino commented on docs/CONTRIBUTING.md:
```diff
@@ -13,56 +16,50 @@ The sections below outline the steps in each case.
+   workflow. Wait until [PyPi publish
+   workflow](https://github.com/WUR-AI/diffwofost/actions/workflows/python-publish.yml)
+   has completed and verify new release is on
+   [PyPi](https://pypi.org/project/matchms/#history)
```
Is the hook that makes a snapshot on Zenodo  also activated? Maybe it could be mentioned here - or maybe not, since there is no action needed by the user.

fnattino commented on src/diffwofost/physical_models/crop/root_dynamics.py:
```diff
@@ -134,13 +127,22 @@ class StateVariables(StatesTemplate):
+                arrays or scalars. See PCSE documentation for details.
+
+        Returns:
+            None: This method initializes the model and does not return anything.
```
Maybe this can be removed (and similarly the returns for `calc_rates` and `integrate`)?

```suggestion
```

fnattino commented on docs/examples.md:
```diff
@@ -0,0 +1,26 @@
+
+We provide an example notebook showing optimization of models' parameters with
+`diffWOFOST`. To get familiar with the concepts and implementation, check out
+`Introduction` in the documentation.
```
Maybe add a link to the first document in the section?
```suggestion
[`Introduction`](./index.md) in the documentation.
```

SarahAlidoost commented on docs/CONTRIBUTING.md:
```diff
@@ -13,56 +16,50 @@ The sections below outline the steps in each case.
-Also a Zenodo entry will be made for the release with its own DOI.
+1. Check the author list in `CITATION.cff`in
+   the root of the repository.
+2. Bump the version. The version can be manually changed in `pyproject.toml` in
```
right, added!

SarahAlidoost commented on docs/project_setup.md:
```diff
@@ -36,55 +37,15 @@ guide](https://guide.esciencecenter.nl/#/best_practices/language_guides/python?i
 
 - We recommend using [semantic versioning](https://guide.esciencecenter.nl/#/best_practices/releases?id=semantic-versioning).
-- For convenience, the package version is stored in a single place: `pyproject.toml` under the `tool.bumpversion` header.
+- For convenience, the package version is stored in a single place: `pyproject.toml`.
```
added!

SarahAlidoost commented:
> Very nice @SarahAlidoost - the documentation pages look great! I have just left a few minor comments, but I think it could be merged at any moment..

Thanks @fnattino. I addressed your comments and suggestions.
